### PR TITLE
[Typescript] Introduce the getGlobalSettings API Method

### PR DESCRIPTION
**Pull request by LastMileNewGrad**
[Typescript] Introduce the getGlobalSettings API Method

This diff introduces the `getGlobalSetting` API method, which retrieves global settings for a specified model.

One of the steps in enhancing ModelParser creation. More in diffs on top.

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/51).
* #76
* #75
* #55
* #52
* #54
* __->__ #51
* #50

### CODE REVIEW

File: typescript/__tests__/testProgramaticallyCreateConfig.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { AIConfigRuntime } from "../lib/config";
+import { InferenceSettings } from "../types";
+
+describe("test extract_override_settings", () => {
```
Comment (saqadri) [inline typescript/__tests__/testProgramaticallyCreateConfig.ts:4]: Update description. Doesn't look correct for this test

File: typescript/__tests__/testProgramaticallyCreateConfig.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  // shared state
+  let aiConfigRuntime: AIConfigRuntime = AIConfigRuntime.create(
+    "Untitled AIConfig",
+    undefined,
+    undefined,
```
Comment (saqadri) [inline typescript/__tests__/testProgramaticallyCreateConfig.ts:9]: When adding "undefined" as a value, add /*param*/ as a comment on that line to describe which param is undefined.
`/*param*/ description`

Also can you double check if this is correct? Looks like you are setting schemaVersion to undefined? Just set it to "latest"

File: typescript/__tests__/testProgramaticallyCreateConfig.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    { models: { testmodel: { topP: 0.9 } } }
+  );
+
+  it("Test Case 1: Retrieve global setting from AIConfig with 1 model", () => {
```
Comment (saqadri) [inline typescript/__tests__/testProgramaticallyCreateConfig.ts:13]: nit: change from `it` to `test`. They are the same.

Also update the description:
`test("Retrieving global settings from AIConfig with 1 model")`.
Reson: in Jest the test names are supposed to read like a proper sentence. In this case: "test retrieving global settings from AIConfig with 1 model"

If you use `it` is can be `it("Retrieves global settings from AIConfig with 1 model")` so the sentence reads "it retrieves global settings from AIConfig with 1 model"

**Review by saqadri — APPROVED** [review]

File: typescript/__tests__/testProgramaticallyCreateConfig.ts (hunk 1 above)
Comment (LastMileNewGrad) [reply typescript/__tests__/testProgramaticallyCreateConfig.ts]: Updated to `test Get Global Settings` to accurately describe which method its testing

File: typescript/__tests__/testProgramaticallyCreateConfig.ts (hunk 2 above)
Comment (LastMileNewGrad) [reply typescript/__tests__/testProgramaticallyCreateConfig.ts]: updated and used `latest` as suggested

File: typescript/__tests__/testProgramaticallyCreateConfig.ts (hunk 3 above)
Comment (LastMileNewGrad) [reply typescript/__tests__/testProgramaticallyCreateConfig.ts]: that's a cool style guideline. Is there a javascript/typescript style guidelines like Python's PEP?

**Pull request merged**